it/v2/updateUKSavingsAccountName/UpdateUKSavingsAccountNameControllerISpec.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.updateUKSavingsAccountName
+
+import com.github.tomakehurst.wiremock.stubbing.StubMapping
+import models.errors.{AccountNameFormatError, SavingsAccountIdFormatError}
+import play.api.http.Status._
+import play.api.libs.json.{JsObject, JsValue, Json}
+import play.api.libs.ws.{WSRequest, WSResponse}
+import play.api.test.Helpers.{ACCEPT, AUTHORIZATION, NO_CONTENT}
+import shared.models.errors._
+import shared.services.{AuditStub, AuthStub, DownstreamStub, MtdIdLookupStub}
+import shared.support.IntegrationBaseSpec
+import v2.updateUKSavingsAccountName.fixture.UpdateUKSavingsAccountNameFixtures.validRequestJson
+
+class UpdateUKSavingsAccountNameControllerISpec extends IntegrationBaseSpec {
+
+  "calling the 'update uk savings account name' endpoint" should {
+    "return a 204 status code" when {
+      "any valid request is made" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+          DownstreamStub.onSuccess(DownstreamStub.PUT, downstreamUri, NO_CONTENT, JsObject.empty)
+        }
+
+        val response: WSResponse = await(request().put(validRequestJson))
+        response.status shouldBe NO_CONTENT
+        response.body shouldBe ""
+      }
+    }
+
+    "return error according to spec" when {
+      val emptyRequestJson: JsValue = JsObject.empty
+
+      val nonValidRequestBodyJson: JsValue = Json.parse(
+        """
+          |{
+          |   "accountName": "Shares savings account!"
+          |}
+        """.stripMargin
+      )
+
+      "validation error" when {
+        def validationErrorTest(requestNino: String,
+                                requestSavingsAccountId: String,
+                                requestBody: JsValue,
+                                expectedStatus: Int,
+                                expectedBody: MtdError,
+                                errorWrapper: Option[ErrorWrapper]): Unit = {
```

Comment:
`errorWrapper` is not used so can be removed as all the test cases have it as None. Please also remove `expectedBodyJson` and just use the expectedBody directly, thanks